lib/internal/http2/compat.js

```diff
@@ -850,29 +849,14 @@ class Http2ServerResponse extends Stream {
   writeEarlyHints(hints) {
     validateObject(hints, 'hints');
 
-    const headers = { __proto__: null };
-
-    const linkHeaderValue = validateLinkHeaderValue(hints.link);
-
-    for (const key of ObjectKeys(hints)) {
-      if (key !== 'link') {
-        headers[key] = hints[key];
-      }
-    }
-
-    if (linkHeaderValue.length === 0) {
-      return false;
```

Comment:
I understand the need to remove `regexp` but why don't we keep the primitive checks as it is?

Reply:
This check skips sending `103 Early Hints` all together if no link header is present. Which would be confusing if you're trying to send early hints with some other header. Admittedly I'm not aware of any current need to send `103 Early Hints` without a link header, but by my reading the [RFC](https://www.rfc-editor.org/rfc/rfc8297.html) doesn't mandate the link header in any way.